CLOUD_MIGRATION_PLAN.md

```diff
@@ -0,0 +1,969 @@
+| 1 × 4GB worker | $0.036/hr | ~$13/month |
+| Mixed (1×2GB + 1×4GB) | $0.048/hr | ~$18/month |
+
+**Total Estimated Cost**: $30-35/month
+
+**Cost Optimization Strategies**:
+1. Scale down worker pool during off-peak hours
+2. Use smaller workers for light workloads
+3. Temporarily add large workers only for heavy jobs
+4. Monitor memory requirements and right-size workers
+
+## Testing Strategy
+
+### Unit Tests
+- Test S3IO with mocked S3 client
+- Test compression round-trip
+- Test archive listing/filtering
+
+### Integration Tests
+- End-to-end cloud worker test (LocalStack for S3)
+- Test with various input sizes
+- Test failure scenarios (S3 unavailable, etc.)
+
+### Acceptance Tests
+- Deploy test droplet with sample input
+- Verify archive uploaded correctly
+- Download and verify archive contents
+
+## Success Criteria
+
+1. **Functional**:
+   - ✓ Workers run as long-lived processes in a pool
+   - ✓ Workers claim files via S3 move (no duplicate work)
+   - ✓ Archives selected using smallest/largest logic (like current)
+   - ✓ Processing uses local disk only (no network during processing)
+   - ✓ Archive uploaded to S3 on completion
+   - ✓ TUI shows per-worker metrics correctly
+   - ✓ Memory requirements enforced at startup
+
+2. **Recovery**:
+   - ✓ Heartbeat updated during processing (to S3)
+   - ✓ Stale heartbeats detected (10 minute grace period)
+   - ✓ Abandoned files moved back to input for re-processing
+   - ✓ Crashed workers' work is not lost
+
+3. **Operations**:
+   - ✓ Workers can be added/removed manually
+   - ✓ Workers can be resized for memory requirements
+   - ✓ Per-worker TUI monitoring available
+
+4. **Cost**:
+   - ✓ No database fees (S3 only)
+   - ✓ Pool can be scaled down when idle
+   - ✓ Right-size workers to actual memory needs
+
+## Timeline Estimate
+
+- **Phase 1** (S3 I/O): 3-5 days
+- **Phase 2** (Main entry point): 2-3 days
+- **Phase 3** (Compression): 1-2 days
+- **Phase 4** (Deployment): 2-3 days
+
+**Total Estimated Time**: 2-3 weeks
+
+## Security Considerations
+
+### Credentials Management
+
+1. **S3 Access Keys**:
+   - Use minimum permissions (read input, write archives)
+   - Inject via environment variables only
+   - Never commit to repository
+
+2. **Droplet Security**:
+   - Use private VPC if available
+   - Firewall: block all inbound traffic
+   - Auto-terminate after timeout
+
+### Data Protection
+
+1. **At Rest**: S3 encryption by default
+2. **In Transit**: HTTPS for all S3 operations
+3. **Local Disk**: Ephemeral (lost on shutdown)
+
+## Conclusion
+
+This migration plan enables distributed processing with a **worker pool model** that provides:
+
+- **Per-worker monitoring**: TUI dashboard shows each worker's metrics (RAM, queue depth, progress)
+- **Dynamic sizing**: Workers can be resized based on runtime memory requirements from `MemoryConfig`
+- **Local recovery**: Heartbeat-based crash recovery within each worker (not cross-worker coordination)
```

Comment:
Remove local recovery from the design. Recovery should be s3 based across the board